packages/integrations/solid/README.md

```diff
@@ -98,6 +98,73 @@ export default defineConfig({
 });
 ```
 
+## Rendering strategy
+
+Hydrating SolidJS components are automatically wrapped in Suspense boundaries and rendered on the server using the [renderToStringAsync](https://www.solidjs.com/docs/latest/api#rendertostringasync) function. This means that lazy components will be resolved and rendered on the server. For example the following source files:
+
+```tsx
+// HelloAstro.tsx
+export default function HelloAstro() {
+  return <div>Hello Astro</div>;
+}
+```
+
+```tsx
+// LazyHelloAstro.tsx
+import { lazy } from 'solid-js';
+export const LazyHelloAstro = lazy(() => import('./HelloAstro'));
+```
+
+<!-- prettier-ignore -->
+```astro
+// hello.astro
+import { LazyHelloAstro } from './LazyHelloAstro.tsx';
+---
+<LazyHelloAstro />
+```
+
+Will be rendered into the server's HTML output as
+
+```html
+<div>Hello Astro</div>
+```
+
+Resources will also be resolved. For example:
```

Comment:
... continuing from the idea in my earlier comment,

For example, in this case, I might suggest something like:

> Resources will also be resolved. For example, a function that awaits a data fetch will be able to render the returned information:

or 

> Resources will also be resolved. The following example shows a function that awaits a data fetch, and then renders the returned information:

(I'm not familiar with Solid, so it's your job to make sure the information is correct. But, we do like to pre-load our explanations so that people know what they should be looking at/for when you put a code sample in front of them.)

I would also argue that the three snippets here are more distracting than helpful. It takes work to figure out what is going on when I think it's kind of assumed that if you write a Solid component, you'll render in a `.astro` file somewhere, and then it will render to HTML. It's probably ok to completely skip the middle code here, because I think what you're really trying to say is "a component written like this will render to this HTML". 

In fact, if *all* you were trying to say is that it would successfully return the data from the fetch, I would argue that you wouldn't even need the final HTML snippet at all. I think it's pretty clear from the function alone that you're *trying* to return `<div>Name: [some actual name]</div>` , so if the key point is just "this works" then I don't think we need three whole code snippets to say that.

Reply:
Addressed in https://github.com/withastro/astro/pull/6791/commits/5a810a9b73463d93b3b7eeee92ea1dd1ca9f7111